sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -1028,4 +1028,21 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(Contains(Literal("Spark SQL"), Literal("SQL")), true)
     checkEvaluation(Contains(Literal("Spark SQL"), Literal("k S")), true)
   }
+
+  test("SPARK-37582, SPARK-37583: CONTAINS, STARTSWITH, ENDSWITH should support binary type") {
+    checkEvaluation(Contains(Literal("Spark SQL"), Literal("Spark")), true)
+    checkEvaluation(Contains(Literal("Spark SQL".getBytes), Literal("Spark")), true)
+    checkEvaluation(Contains(Literal("Spark SQL"), Literal("Spark".getBytes)), true)
```

Comment:
What is the purpose of those checkEvaluation()? How do they check your changes?

I think need more tests for your new functions `contains`, `startsWith`, `endsWith`, and especially `indexOf` to check corner cases.

Reply:
`Literal("Spark".getBytes)` will create `BinaryType` then can check new method support `BinaryType`